### DISCUSSION TITLE
How many tokens would an array take up?

### DISCUSSION THREAD

**Discussion by Georgej100** [SebLague/Chess-Challenge#123] [Q&A] [answered]
In the documentation it says that a variable are counted as 1 token. I was just wondering whether an array would use 1 token or its length?

**Comment by SebLague** [comment, marked as answer]
If you write `int[] array = { 1, 2, 3 };` for example it will count as 9 tokens. The breakdown of counted tokens is:
IntKeyword  int
OpenBracketToken  [
OmittedArraySizeExpressionToken (hmm I should maybe remove this one, I didn't notice this before)
IdentifierToken  array
EqualsToken  =
OpenBraceToken  {
NumericLiteralToken  1
NumericLiteralToken  2
NumericLiteralToken  3

**Comment by SebLague** [comment]
There's a line in the TokenCounter script `Console.WriteLine(kind + "  " + syntaxNode.ToString());` that you can uncomment if you'd like to see this breakdown for your own code.

**Reply by Georgej100** [reply]
Thanks!
